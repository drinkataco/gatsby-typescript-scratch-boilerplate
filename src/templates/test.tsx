import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';

import MainTemplate from './main';

describe('main content template', () => {
  const props = {
    data: {
      markdownRemark: {
        id: 'test-123',
        html: '<p>test</p>',
        frontmatter: {
          title: 'Test Markdown Page',
          description: 'Test Markdown Page Description',
        },
      },
    },
  };

  beforeEach(() => {
    (StaticQuery as jest.Mock).mockImplementationOnce(
      ({ render }) => render(props),
    );
    (useStaticQuery as jest.Mock).mockReturnValue({ site: { siteMetadata: {} } });
  });

  it('renders with correct data', () => {
    expect.assertions(1);

    const tree = renderer.create(<MainTemplate data={props.data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
