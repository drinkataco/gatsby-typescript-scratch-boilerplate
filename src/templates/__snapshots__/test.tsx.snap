// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main content template renders with correct data 1`] = `
<div
  className="layout"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
